samples/bluetooth/hci_uart/src/main.c

```diff
@@ -136,83 +206,20 @@ static void tx_isr(void)
 
 static void bt_uart_isr(const struct device *unused, void *user_data)
 {
-	static struct net_buf *buf;
-	static int remaining;
-
 	ARG_UNUSED(unused);
 	ARG_UNUSED(user_data);
 
-	while (uart_irq_update(hci_uart_dev) &&
-	       uart_irq_is_pending(hci_uart_dev)) {
-		int read;
-
-		if (uart_irq_tx_ready(hci_uart_dev)) {
-			tx_isr();
-		} else if (!uart_irq_rx_ready(hci_uart_dev)) {
-			/* Only the UART TX and RX path are interrupt-enabled */
-			LOG_DBG("spurious interrupt");
-			break;
-		}
-
-		/* Beginning of a new packet */
-		if (!remaining) {
-			uint8_t type;
-
-			/* Get packet type */
-			read = h4_read(hci_uart_dev, &type, sizeof(type), 0);
-			if (read != sizeof(type)) {
-				LOG_WRN("Unable to read H4 packet type");
-				continue;
-			}
-
-			buf = bt_buf_get_tx(BT_BUF_H4, K_NO_WAIT, &type,
-					    sizeof(type));
-			if (!buf) {
-				return;
-			}
-
-			switch (bt_buf_get_type(buf)) {
-			case BT_BUF_CMD:
-				h4_cmd_recv(buf, &remaining);
-				break;
-			case BT_BUF_ACL_OUT:
-				h4_acl_recv(buf, &remaining);
-				break;
-			default:
-				LOG_ERR("Unknown H4 type %u", type);
-				return;
-			}
-
-			LOG_DBG("need to get %u bytes", remaining);
-
-			if (remaining > net_buf_tailroom(buf)) {
-				LOG_ERR("Not enough space in buffer");
-				net_buf_unref(buf);
-				buf = NULL;
-			}
-		}
-
-		if (!buf) {
-			read = h4_discard(hci_uart_dev, remaining);
-			LOG_WRN("Discarded %d bytes", read);
-			remaining -= read;
-			continue;
-		}
-
-		read = h4_read(hci_uart_dev, net_buf_tail(buf), remaining, 0);
-
-		buf->len += read;
-		remaining -= read;
-
-		LOG_DBG("received %d bytes", read);
+	if (!(uart_irq_rx_ready(hci_uart_dev)
+	      || uart_irq_tx_ready(hci_uart_dev))) {
```

Comment:
|| should be at the end of the line above.

Reply:
done